### ISSUE TITLE
Error when changing scenes with packed Scene

### ISSUE THREAD

**Issue by devmark404** [godotengine/godot#82757]
### Godot version

4.1.1

### System information

Window 10 - Godot v4.1.1 - Mobile

### Issue description

Scenes get corrupted after trying to change scenes

Show this errors

 scene/resources/resource_format_text.cpp:283 - res://scene_02.tscn:20 - Parse Error: 
  Failed loading resource: res://scene_02.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  scene/resources/resource_format_text.cpp:283 - res://scene_01.tscn:20 - Parse Error: 
  Failed loading resource: res://scene_01.tscn. Make sure resources have been imported by opening the project in the editor at least once.

Video
https://youtu.be/42Z1KEkwKwY

### Steps to reproduce

I have a button with an export variable to change scenes

```
@export var change_scene:PackedScene

func _ready() -> void:
	pass

func _on_pressed() -> void:
	get_tree().change_scene_to_packed(change_scene)
	pass

```

If I then try to use that button within a scene to switch to another scene, the scenes become corrupted and can no longer be opened

I even made a copy of the project in another folder before adding the scenes to the export variable, but that project was also damaged

[ErrorsChangeScene.zip](https://github.com/godotengine/godot/files/12798254/ErrorsChangeScene.zip)

Video
https://youtu.be/42Z1KEkwKwY

### Minimal reproduction project

Video
https://youtu.be/42Z1KEkwKwY

**Comment by theraot** [comment]
From testing, it seem that the problem is that the scenes create a cyclic dependency. Editing either of the scene to remove the reference allows both scenes to load.

**Comment by devmark404** [comment]
I just tried the same thing with the new godot 4.1.2 update and the same thing happens

Here I leave the project before the error occurs

Just drag scene 2 onto the scene 1 button
And drag scene 1 onto the scene 2 button
Then save the scenes and restart the project

https://mega.nz/file/JfwhVKiS#-kVVZ8O4gEK0bcwF5XjF5UZiQSW0vABGwX24R6sLnBc

-Español
Acabo de probar lo mismo en con la nueva actualización de godot 4.1.2 y pasa lo mismo

Aquí dejo el proyecto antes de que ocurra el error

Solo arrastra la escena 2 en el botón de la escena 1
Y arrastra la escena 1 en el botón de la escena 2
Después guarda las escenas y reinicia el proyecto
https://mega.nz/file/JfwhVKiS#-kVVZ8O4gEK0bcwF5XjF5UZiQSW0vABGwX24R6sLnBc

**Comment by sulaemansantoso** [comment]
> From testing, it seem that the problem is that the scenes create a cyclic dependency. Editing either of the scenes to remove the reference to the other allows both scenes to load.

This is true, you can either do a clear inheritance on the inherited  scene or just use change_scene with string parameter
